datastore/index.go

```diff
@@ -0,0 +1,85 @@
+// Datastore implements workers that work with the Redis Index and Cassandra Metric datastores
+package datastore
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"strconv"
+	"strings"
+
+	"github.com/jeffpierce/cassabon/config"
+	"github.com/jeffpierce/cassabon/middleware"
+	"gopkg.in/redis.v3"
+)
+
+// IndexMetricPath takes a metric path string, splits it, and sends it off to be recursively parsed and indexed.
+func IndexMetricPath(path string) {
+	// We've got a path!
+	config.G.Log.System.LogDebug("Indexer received path %s.")
+
+	// Split the metric into a slice of its parts
+	splitPath := strings.Split(path, '.')
+
+	// Initialize a Redis client.
+	if config.G.Redis.Index.Sentinel {
+		config.G.Log.System.LogDebug("Initializing Redis client (Sentinel)")
+		rc := middleware.RedisFailoverClient(
+			config.G.Redis.Index.Addr,
+			config.G.Redis.Index.Pwd,
+			config.G.Redis.Index.Master,
+			config.G.Redis.Index.DB,
+		)
+	} else {
+		config.G.Log.System.LogDebug("Initializing Redis client...")
+		rc := middleware.RedisClient(
+			config.G.Redis.Index.Addr,
+			config.G.Redis.Index.Pwd,
+			config.G.Redis.Index.DB,
+		)
+	}
+
+	// Ship the metric and a Redis pipeline to the metric processor.
+	pipe := rc.Pipeline()
```

Comment:
"defer pipe.Close()" right here will close it even if something further down panics.